clouds/clouds-azure/src/main/java/azure/core/AzureUtils.java

```diff
@@ -50,12 +49,68 @@ static Optional<VirtualMachine> findAzureVM(AzureResourceManager arm, String nam
                 .findAny();
     }
 
-    static Optional<WebAppBasic> findAzureWebApp(AzureResourceManager arm, String name, String resourceGroup) {
-        return arm.webApps().listByResourceGroup(resourceGroup) .stream()
-                .filter(webApp -> webApp.name().equals(name))
-                .findAny();
+    static Optional<WebApp> findAzureWebApp(AzureResourceManager arm, String name, String resourceGroup) {
+        try {
+            return Optional.ofNullable(arm.webApps().getByResourceGroup(resourceGroup, name));
+        } catch (Exception e) {
+            e.printStackTrace();
+            return Optional.empty();
+        }
+    }
+
+    static void waitForWebAppDeployment(WebApp app) throws InterruptedException, IOException {
+        OkHttpClient client = new OkHttpClient();
+        // 20 minutes in millis
+        long timeout = TimeoutUtils.adjust(1200000);
+        long start = System.currentTimeMillis();
+        while (System.currentTimeMillis() - start < timeout) {
+            try {
+                Response response = client.newCall(new Request.Builder()
+                                .url("http://" + app.defaultHostname())
+                                .method("GET", null)
+                                .build())
+                        .execute();
+                int code = response.code();
+                String body = response.body() != null ? response.body().string() : null;
+                if (code < 500 && !isWebAppDnsProblem(code, body) && !isEapWebAppWelcomePage(code, body)) {
```

Comment:
Checks in `isWebAppDnsProblem(code, body) && !isEapWebAppWelcomePage` seems to be quite squishy and depend on HTML body/content. Does Azure provide some API for this?

Reply:
I am not aware of any. I will ping Denis from MS team if he knows something.

Reply:
Denis is not very responsive. Once we will get more info, I will happily integrated suggested solution.